account_financial_report/report/open_items.py

```diff
@@ -119,19 +121,27 @@ def _get_data(
             journals_ids.add(move_line["journal_id"][0])
             acc_id = move_line["account_id"][0]
             # Partners data
-            if move_line["partner_id"]:
-                prt_id = move_line["partner_id"][0]
-                prt_name = move_line["partner_id"][1]
+            partner_id = move_line.get("partner_id")
+            if partner_id:
+                partner = self.env["res.partner"].browse(partner_id[0])
+                user_id = partner.user_id
+                if grouped_by == "salesperson":
+                    prt_id = user_id.id if user_id else 0
+                    prt_name = user_id.name if user_id else _("Missing Salesperson")
+                else:
```

Comment:
Maybe others extend the module and you don't want a general `else`.

```suggestion
                elif grouped_by == "partners":
```

Reply:
If i change this, pre-commit fails
account_financial_report/report/open_items.py:62:5: C901 'OpenItemsReport._get_data' is too complex (17)
Is a way to ignore it?

Reply:
I have simplified the code in https://github.com/OCA/account-financial-reporting/compare/1e8070a5cdea81839780db9d5c1faa0086294ba1..19d84722900eb520201e622ae5fac35400e600c3#diff-6ecb2fc7ae08d3de3133f69e4b7616d0e6715b2fb907b4d84f02a43fed43c481